src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -0,0 +1,206 @@
+                {
+                    IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+
+                    if (splittingStrategy != SplittingStrategy.PerRun)
+                    {
+                        SplittingVisitor visitor;
+                        switch (splittingStrategy)
+                        {
+                            case SplittingStrategy.PerRunPerRule:
+                                visitor = new PerRunPerRuleSplittingVisitor();
+                                break;
+                            
+                            case SplittingStrategy.PerRunPerTarget:
+                                visitor = new PerRunPerTargetSplittingVisitor();
+                                break;
+                            
+                            case SplittingStrategy.PerRunPerTargetPerRule:
+                                visitor = new PerRunPerTargetPerRuleSplittingVisitor();
+                                break;
+
+                            // TODO: Implement PerResult and PerRun splittings strategies
+                            //
+                            //       https://github.com/microsoft/sarif-sdk/issues/1763
+                            //       https://github.com/microsoft/sarif-sdk/issues/1762
+                            //
+                            case SplittingStrategy.PerResult:
+                            case SplittingStrategy.PerRun:                           
+                            default:
+                                throw new ArgumentOutOfRangeException($"GroupingStrategy: {splittingStrategy}");
+                        }
+
+                        visitor.VisitRun(sarifLog.Runs[runIndex]);
+                        logsToProcess = visitor.SplitSarifLogs;
+                    }
+
+                    for (int splitFileIndex = 0; splitFileIndex < logsToProcess.Count; splitFileIndex++)
+                    {
+                        SarifLog splitLog = logsToProcess[splitFileIndex];
+                        FileWorkItemsForSarifLog(sarifLog);
+                    }
+                }
+            }
+        }
+
+        private void FileWorkItemsForSarifLog(SarifLog sarifLog)
+        {
+            // First we make a copy of the pipeline root context. This allows us to produce
+            // new context files, potentially with outputs, that are specific to each log.
+            
+            SarifWorkItemModel workItemModel = sarifLog.CreateWorkItemModel(this.FilingContext);
+
+            // Now we initialize this work item model from the SARIF log file driving the
+            // current operations. As a first step, this code should extract all 
+            // source-controlled files from the log file. In an imminent implementation
+            // this information will be used to derive area path and owner information.
+            
+            workItemModel.Context.InitializeFromLog(sarifLog);
```

Comment:
Is this not already done if you follow the above extension method through into model?

Reply:
+1. Yes, it is.

---
In reply to: [380979633](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r380979633) [](ancestors = 380979633)

Reply:
Yes, and the comments are bad, thanks.

---
In reply to: [380979633](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r380979633) [](ancestors = 380979633)